pkg/oc/cli/cmd/newapp.go

```diff
@@ -373,6 +374,10 @@ func (o *NewAppOptions) RunNewApp() error {
 				hasMissingRepo = true
 				fmt.Fprintf(out, "%sWARNING: No Docker registry has been configured with the server. Automatic builds and deployments may not function.\n", indent)
 			}
+		case *routeapi.Route:
+			if len(t.Spec.Host) > 0 {
+				fmt.Fprintf(out, "%sAccess your application via route '%s' \n", indent, t.Spec.Host)
```

Comment:
If we find a route object it's possible Host will be empty because the template.Route didn't specify a value. (which means it'll get filled in by the server when the Route object is created).  This is where we should really be polling the created object on the server until it has a valid Host value filled in, and then report that value (similar to how your added readiness check works).

Reply:
Does this mean we will create blocking process here? Should we poll the server for let's say 5s and if we cant get it we either error or dont print it at all?

Reply:
yes i think a cap on how long to wait is a good idea.

Reply:
@jim-minter @bparees do you have any guide how I can get client for this to do this.
I was thinking to add something like 
```
// Get object to check status
func Get(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
	if len(info.Namespace) > 0 {
		namespace = info.Namespace
	}
	return resource.NewHelper(info.Client, info.Mapping).Get(namespace, info.Name, false)
}
```
to cmd.go

but at this point I'm a little bit lost about all `b.Op, bRun, etc etc` I might be looking to this from the wrong angle...

Reply:
What about this? Sorry for stupid question. Stil trying to get my head around on different implementations styles around the codebase. 
```
// getRouteHost will poll server for host field to be populated. With timout
func getRouteHost(c *newcmd.AppConfig, r *routeapi.Route) (*routeapi.Route, error) {
	var route *routeapi.Route
	c1 := make(chan string, 1)
	go func() {
		time.Sleep(time.Second * RoutePollTimoutSeconds)
		c1 <- "Route creation timout"
	}()

	select {
	case res := <-c1:
		return nil, err.New(res)
	case <-time.After(time.Microsecond * 500):
		route, err := c.OSClient.Routes(r.Namespace).Get(r.Name, metav1.GetOptions{})
		if err != nil {
			return nil, err
		}
		if route.Spec.Host != "" {
			return route, nil
		}
	}
	return route, nil
}
```

Reply:
check out wait.Poll as it's used elsewhere in the codebase.

Reply:
done